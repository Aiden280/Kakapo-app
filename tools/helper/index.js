import fs from 'fs-extra';

// Upload stats.json to `http://webpack.github.io/analyse` for analysis.
// Also you can analyze size with `https://github.com/robertknight/webpack-bundle-size-analyzer`:
// `cat stats.json | analyze-bundle-size`
export const writeStats = stats =>
  fs.writeFile('stats.json', JSON.stringify(stats.toJson(), null, 4));

export default {};
