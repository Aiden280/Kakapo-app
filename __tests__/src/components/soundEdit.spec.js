import React from 'react';
import { shallow } from 'enzyme';
import { notifyActions, soundActions } from 'actions/';
import { SoundEdit } from 'components/';
import TextInput from 'components/ui/textInput/textInput';
import { newSoundObj } from 'utils/';
import { getData, mockEvent } from '../helper';

function setup(props = {}) {
  const propData = {
    ...getData('themes'),
    ...getData('intl'),
    soundActions: {},
    ...props
  };
  return { props, wrapper: shallow(<SoundEdit {...propData} />) };
}

const soundProp = (props = {}) => {
  const obj = { ...newSoundObj, source: 'file', progress: 1, ...props };
  return {
    sound: Object.keys(obj).reduce((newObj, _e) => {
      newObj[_e] = obj[_e] === null ? 'wind' : obj[_e];
      return newObj;
    }, {})
  };
};

const sound = { ...soundProp() };

test('<SoundEdit/> render', () => {
  const { wrapper } = setup(sound);
  expect(wrapper.type()).toBe('div');
  expect(wrapper.prop('className')).toBe('item editing');
});

test('<SoundEdit/> render 2 inputs', () => {
  const { wrapper } = setup(sound);
  expect(wrapper.find(TextInput).length).toBe(2);
});

test('<SoundEdit/> handleCancel', () => {
  const dispatch = jest.fn();
  const action = soundActions.edit(sound, null);
  const { wrapper } = setup({ sound, dispatch });
  wrapper.find('.button').at(0).simulate('click', mockEvent);
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0]).toEqual(action);
});

test('<SoundEdit/> handleSubmit empty', () => {
  const dispatch = jest.fn();
  const action = notifyActions.send('Please fill out all fields');
  const { wrapper } = setup({ sound, dispatch });
  wrapper.find('form').simulate('submit', {
    ...mockEvent,
    target: { getElementsByTagName: () => ({}) }
  });
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0].type).toBe(action.type);
  expect(dispatch.mock.calls[0][0].data).toBe(action.data);
});

test('<SoundEdit/> handleSubmit filled', () => {
  const dispatch = jest.fn();
  const action = soundActions.edit(sound, { name: 'hi' });
  const { wrapper } = setup({ sound, dispatch });
  wrapper.find('form').simulate('submit', {
    ...mockEvent,
    target: { getElementsByTagName: () => [{ name: 'name', value: 'hi' }] }
  });
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0]).toEqual(action);
});
