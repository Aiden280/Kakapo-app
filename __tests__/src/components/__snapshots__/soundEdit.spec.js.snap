// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundEdit/> render 1`] = `
<div
  className="item editing"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <input
        autoComplete="off"
        defaultValue="wind"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <span
        className="highlight"
      />
      <span
        className="bar"
      />
      <label>
        Name
      </label>
    </div>
    <div
      className="group"
    >
      <input
        autoComplete="off"
        defaultValue="wind"
        name="tags"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <span
        className="highlight"
      />
      <span
        className="bar"
      />
      <label>
        Tag
      </label>
    </div>
    <a
      className="button"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      Cancel
    </a>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "#4CAF50",
          "borderColor": "#4CAF50",
        }
      }
    >
      Save
    </button>
  </form>
</div>
`;
